utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,205 @@
+            identifier.value = rootVal;
+        }
+        return identifier;
+    } else {
+        log:printDebug("Mandatory field root not available", rootVal);
+        return ();
+    }
+}
+
+# Map C-CDA address to FHIR Address.
+#
+# + addressElement - C-CDA address element
+# + return - Return FHIR Address
+public isolated function mapCcdaAddressToFhirAddress(xml addressElement) returns r4:Address? {
+    r4:Address address = {};
+    xml streetAddressLineElement = addressElement/<v3:streetAddressLine|streetAddressLine>;
+    xml cityElement = addressElement/<v3:city|city>;
+    xml stateElement = addressElement/<v3:state|state>;
+    xml postalCodeElement = addressElement/<v3:postalCode|postalCode>;
+
+    string streetAddressLine = streetAddressLineElement.data().trim();
+    string city = cityElement.data().trim();
+    string state = stateElement.data().trim();
+    string postalCode = postalCodeElement.data().trim();
+
+    if streetAddressLine != "" || city != "" || state != "" || postalCode != "" {
+        address.line = [streetAddressLine];
+        address.city = city;
+        address.state = state;
+        address.postalCode = postalCode;
+        return address;
+    } else {
+        log:printDebug("Address fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA telecom to FHIR ContactPoint.
+#
+# + telecomElement - C-CDA telecom element
+# + return - Return FHIR ContactPoint
+public isolated function mapCcdaTelecomToFhirTelecom(xml telecomElement) returns r4:ContactPoint? {
+    r4:ContactPoint telecom = {};
+    string?|error telecomUse = telecomElement.use;
+    string?|error telecomValue = telecomElement.value;
+
+    if telecomValue is string {
+        telecom.value = telecomValue;
+    } else {
+        log:printDebug("Telecom value not available", telecomValue);
+    }
+
+    if telecomUse is string {
+        match telecomUse {
+            "HP" => {
+                telecom.use = "home";
+            }
+            "WP" => {
+                telecom.use = "work";
+            }
+        }
+    } else {
+        log:printDebug("Telecom use not available", telecomUse);
+    }
+
+    if telecom.use is string && telecom.value is string{
+        return telecom;
+    } else {
+        log:printDebug("telecom fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA name to FHIR HumanName.
+#
+# + nameElement - C-CDA name element
+# + return - Return FHIR HumanName
+public isolated function mapCcdaNametoFhirName(xml nameElement) returns r4:HumanName? {
+    r4:HumanName name = {};
+    xml familyElement = nameElement/<v3:family|family>;
+    xml givenElement = nameElement/<v3:given|given>;
+
+    string givenName = givenElement.data().trim();
+    string family = familyElement.data().trim();
+
+    if givenName != "" || family != "" {
+        name.given = [givenElement.data()];
+        name.family = familyElement.data();
+        return name;
```

Comment:
Why is name a variable defined outside?

```suggestion
        return {given: [givenElement.data()], family: familyElement.data()};
```

Reply:
Recheck wherever applicable. This pattern and early returns.

Reply:
Fixed in applicable places in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/cc257cbf67919d7d86a4e90e88ec03e1250eadcf